Sources/ImperialCore/Helpers/String+Tools.swift

```diff
@@ -0,0 +1,23 @@
+import Foundation
+import RoutingKit
+
+extension String {    
+	var pathComponents: [PathComponent] {
+		var pathComponentArray = [PathComponent]()
+		if let components = URL(string: self)?.pathComponents {
+			for item in components where item != "/" {
+				pathComponentArray.append(PathComponent(stringLiteral: item))
+			}
+		}
```

Comment:
I would probably write this:

```swift
return URL(string: self)?.pathComponents.compactMap { component -> PathComponent? in
    guard component != "/" else { return nil }
    return PathComponent(stringLiteral: component)
} ?? []
```
But if you want to leave it, go ahead. It's just a style difference.